crates/sui-json-rpc/src/read_api.rs

```diff
@@ -71,8 +71,27 @@ use fastcrypto::traits::ToFromBytes;
 use shared_crypto::intent::Intent;
 use sui_json_rpc_types::ZkLoginVerifyResult;
 use sui_types::authenticator_state::{get_authenticator_state, ActiveJwk};
+
+/// A field access in a  Display string cannot exceed this level of nesting.
 const MAX_DISPLAY_NESTED_LEVEL: usize = 10;
 
+/// Default budget for Display output size.
+const DEFAULT_MAX_DISPLAY_OUTPUT_SIZE: usize = 1024 * 1024;
+
+/// Overall display output cannot exceed this size.
+static MAX_DISPLAY_OUTPUT_SIZE: Lazy<usize> = Lazy::new(|| {
+    let max_opt = std::env::var("MAX_DISPLAY_OUTPUT_SIZE")
+        .ok()
+        .and_then(|s| s.parse().ok());
+
+    if let Some(max) = max_opt {
+        info!("Using custom value for 'MAX_DISPLAY_OUTPUT_SIZE': {max}");
+        max
+    } else {
+        DEFAULT_MAX_DISPLAY_OUTPUT_SIZE
+    }
+});
```

Comment:
As with in other areas this is probably better served in a config vs an envvar. are you planning on having this be a config in graphql? (once we can support display in grpc we'll thread this through config). I understand that this is probably easier to just unblock given jsonrpc's deprecation

Reply:
Yeah, that's right -- this is just to unblock, but in the new GraphQL implementation, it will be a config.